### PR TITLE
feat: update OCIMachineTemplate webhook validations

### LINKED ISSUE #71: Add more validation for OCIMachineTemplate
**What would you like to be added**:
Now that #55 has been merged we need to work on adding more validation checks for things like the following:

- CompartmentId
- Shape is valid (might need to call to OCI for validation or the OCI SDK might have pre-flight validation we could use. This one might not be straight foreword so some thought should be used on this validation via a webhook or not.)
- NetworkDetails (valid CIDRs)
- IsPvEncryptionInTransitEnabled (some bare metal shapes require this to be false https://docs.oracle.com/en-us/iaas/releasenotes/changes/60d602f5-abb3-4639-aa19-292a5744a808/) 

**Why is this needed**:
Now that #55 is merged we should expand upon our validation. This issue isn't to 100% cover all possibilities, but rather cover what we can now that make sense.

### PR DESCRIPTION
**Pull request by joekr**
**What this PR does / why we need it**:
Adding more validation to the `OCIMachineTemplate` webhook

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #71

### CODE REVIEW

**Comment by joekr** [comment]
# unit
```
?       github.com/oracle/cluster-api-provider-oci      [no test files]
ok      github.com/oracle/cluster-api-provider-oci/api/v1beta1  0.284s  coverage: 8.3% of statements
?       github.com/oracle/cluster-api-provider-oci/cloud/config [no test files]
ok      github.com/oracle/cluster-api-provider-oci/cloud/ociutil        0.610s  coverage: 30.3% of statements
ok      github.com/oracle/cluster-api-provider-oci/cloud/scope  176.358s        coverage: 81.9% of statements
?       github.com/oracle/cluster-api-provider-oci/cloud/scope/mocks    [no test files]
?       github.com/oracle/cluster-api-provider-oci/cloud/services/compute       [no test files]
?       github.com/oracle/cluster-api-provider-oci/cloud/services/compute/mock_compute  [no test files]
?       github.com/oracle/cluster-api-provider-oci/cloud/services/identity      [no test files]
?       github.com/oracle/cluster-api-provider-oci/cloud/services/identity/mock_identity        [no test files]
?       github.com/oracle/cluster-api-provider-oci/cloud/services/networkloadbalancer   [no test files]
?       github.com/oracle/cluster-api-provider-oci/cloud/services/networkloadbalancer/mock_nlb  [no test files]
?       github.com/oracle/cluster-api-provider-oci/cloud/services/vcn   [no test files]
?       github.com/oracle/cluster-api-provider-oci/cloud/services/vcn/mock_vcn  [no test files]
ok      github.com/oracle/cluster-api-provider-oci/controllers  26.349s coverage: 70.7% of statements
```

# e2e
```
```

File: api/v1beta1/ocimachinetemplate_webhook.go [hunk 1]
```diff
@@ -94,12 +96,22 @@ func (m *OCIMachineTemplate) validate() field.ErrorList {
 		allErrs = append(
 			allErrs,
 			field.Invalid(
-				field.NewPath("Spec", "Template", "Spec", "compartmentId"),
+				field.NewPath("Spec", "Template", "Spec", "CompartmentId"),
 				m.Spec.Template.Spec.CompartmentId,
```
Comment (shyamradhakrishnan) [inline api/v1beta1/ocimachinetemplate_webhook.go:100]: everything should be small letters here right?

Comment (shyamradhakrishnan) [reply api/v1beta1/ocimachinetemplate_webhook.go:100]: Users always see small letters in their templates

File: api/v1beta1/ocimachinetemplate_webhook_test.go [hunk 2]
```diff
@@ -57,7 +60,23 @@ var tests = []struct {
 				},
 			},
 		},
-		expectErr: true,
+		errorField: "CompartmentId",
+		expectErr:  true,
+	},
+	{
+		name: "shouldn't allow In-transit encryption for bare metal instances",
```
Comment (shyamradhakrishnan) [inline api/v1beta1/ocimachinetemplate_webhook_test.go:67]: the description?

**Comment by joekr** [comment]
after updates

# unit
```
?       github.com/oracle/cluster-api-provider-oci      [no test files]
ok      github.com/oracle/cluster-api-provider-oci/api/v1beta1  0.270s  coverage: 9.1% of statements
?       github.com/oracle/cluster-api-provider-oci/cloud/config [no test files]
ok      github.com/oracle/cluster-api-provider-oci/cloud/ociutil        0.461s  coverage: 30.3% of statements
ok      github.com/oracle/cluster-api-provider-oci/cloud/scope  176.079s        coverage: 81.9% of statements
?       github.com/oracle/cluster-api-provider-oci/cloud/scope/mocks    [no test files]
?       github.com/oracle/cluster-api-provider-oci/cloud/services/compute       [no test files]
?       github.com/oracle/cluster-api-provider-oci/cloud/services/compute/mock_compute  [no test files]
?       github.com/oracle/cluster-api-provider-oci/cloud/services/identity      [no test files]
?       github.com/oracle/cluster-api-provider-oci/cloud/services/identity/mock_identity        [no test files]
?       github.com/oracle/cluster-api-provider-oci/cloud/services/networkloadbalancer   [no test files]
?       github.com/oracle/cluster-api-provider-oci/cloud/services/networkloadbalancer/mock_nlb  [no test files]
?       github.com/oracle/cluster-api-provider-oci/cloud/services/vcn   [no test files]
?       github.com/oracle/cluster-api-provider-oci/cloud/services/vcn/mock_vcn  [no test files]
ok      github.com/oracle/cluster-api-provider-oci/controllers  26.317s coverage: 71.0% of statements
```

# e2e
```
Ran 5 of 21 Specs in 1881.020 seconds
SUCCESS! -- 5 Passed | 0 Failed | 0 Pending | 16 Skipped
```

**Review by shyamradhakrishnan — APPROVED** [review]

**Pull request merged**